Source/Dafny/Compilers/Compiler-Csharp.cs

```diff
@@ -1636,13 +1701,21 @@ protected class ClassWriter : IClassWriter {
         }
         cw.StaticMemberWriter.WriteLine(";");
       } else if (cw.CtorBodyWriter == null) {
-        cw.InstanceMemberWriter.Write($"{publik}{konst} {typeName} {name}");
+        if (isPublic && !isConst) {
+          cw.InstanceMemberWriter.Write($"{publik}{konst} virtual {typeName} {name} {{get; set;}}");
+        } else {
+          cw.InstanceMemberWriter.Write($"{publik}{konst} {typeName} {name}");
+        }
         if (rhs != null) {
           cw.InstanceMemberWriter.Write($" = {rhs}");
         }
         cw.InstanceMemberWriter.WriteLine(";");
       } else {
-        cw.InstanceMemberWriter.WriteLine($"{publik} {typeName} {name};");
+        if (isPublic) {
+          cw.InstanceMemberWriter.Write($"{publik} virtual {typeName} {name} {{get; set;}}");
```

Comment:
I think a read-only property should not have a setter.

Reply:
I tried removing the setter but it causes C# compilation errors. I don't think `virtual` implies read-only, if I understand your concern correctly.